### PR TITLE
clone and use upstream vsp package

### PR DESCRIPTION
**Pull request by itswisdomagain**
The [upstream vsp package](https://github.com/decred/dcrwallet/tree/v2.0.0-rc3/internal/vsp) provides more robust VSP tickets handling and includes provision for retrying VSP fee payments should the initial attempt error.

### CODE REVIEW

**Comment by itswisdomagain** [comment]
[This commit](https://github.com/planetdecred/dcrlibwallet/pull/225/commits/8c72a1714be0448a1ad2495831c1b60d4584e680) introduces a deviation from the upstream spv package. Requires special review to be sure the change doesn't break anything.

File: internal/vsp/log.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+// log is a logger that is initialized with no output filters.  This
+// means the package will not perform any logging by default until the caller
+// requests it.
+var log = slog.Disabled
+
+// UseLogger uses a specified Logger to output package logging info.
+// This should be used in preference to SetLogWriter if the caller is also
+// using slog.
+func UseLogger(logger slog.Logger) {
```
Comment (itswisdomagain) [inline internal/vsp/log.go:17]: Ensure this is called.

**Comment by itswisdomagain** [comment]
This is being tested from https://github.com/planetdecred/godcr/pull/769 and it'll be ideal to ensure that works smoothly before merging this.

**Review by Sirmorrison — APPROVED** [review]

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
PR code base looks good. Just some minor changes.

File: multiwallet.go [hunk 2]
```diff
@@ -43,7 +43,9 @@ type MultiWallet struct {
 
 	Politeia  *Politeia
 	dexClient *DexClient
-	VspList   []*VSPInfo
+
+	vspMu sync.RWMutex
+	vsps  []*VSP
```
Comment (Sirmorrison) [inline multiwallet.go:48]: This does not look like it is used anywhere.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: multiwallet.go (hunk 2 above)
Comment (itswisdomagain) [reply multiwallet.go:48]: artefact, will remove

Comment (itswisdomagain) [reply multiwallet.go:48]: eh, no, they're used in the vsp.go file

**Review by Sirmorrison — APPROVED** [review]
LGTM

**Pull request merged**